### PR TITLE
[feature] retrieve specific smtp certificate. Closes: #4284

### PR DESCRIPTION
**Pull request by kalikaneko**
this is ready, but it needs complete testing.
it currently needs platform:develop, which is deployed in dev.bm

### CODE REVIEW

File: src/leap/bitmask/services/mail/smtpbootstrapper.py [hunk 1]
```diff
@@ -99,9 +102,21 @@ def _download_config_and_cert(self):
...
                 return
 
-            download_client_cert(self._provider_config,
-                                 client_cert_path,
-                                 self._session)
+            try:
+                download_client_cert(self._provider_config,
+                                     client_cert_path,
+                                     self._session, kind="smtp")
+            except HTTPError:
```
Comment (ivanalejandro0) [inline src/leap/bitmask/services/mail/smtpbootstrapper.py]: shouldn't we catch specifically `403`? can we? are there any other error cases that we may supress like this?

Comment (kalikaneko) [inline src/leap/bitmask/services/mail/smtpbootstrapper.py]: you're right. I quickly looked and saw that in the place where the exception is catched we don't have the return code, *but* it is in the exception message.

File: src/leap/bitmask/crypto/certs.py [hunk 2]
```diff
@@ -41,32 +42,46 @@ def download_client_cert(provider_config, path, session):
...
-    logger.debug('getting cert from uri: %s' % cert_uri)
+
+    logger.debug('getting %s cert from uri: %s' % (kind, cert_uri))
 
     headers = {}
 
     # API v2 will only support token auth, but in v1 we can send both
     if token is not None:
-        headers["Authorization"] = 'Token token="{0}"'.format(token)
+        uuid = srp_auth.get_uuid()
```
Comment (ivanalejandro0) [inline src/leap/bitmask/crypto/certs.py]: it seems that we don't use this `uuid` anywhere...

File: src/leap/bitmask/crypto/certs.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 """
 Utilities for dealing with client certs
 """
+import base64
```
Comment (ivanalejandro0) [inline src/leap/bitmask/crypto/certs.py]: I don't think we need this import.

**Pull request merged**